core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2867,10 +2859,11 @@ public PhysicalOperation visitDynamicFilterSource(DynamicFilterSourceNode node,
                             Map.Entry::getKey,
                             entry -> source.getTypes().get(entry.getValue())));
 
+            TaskContext taskContext = context.getTaskContext();
             LocalDynamicFilterConsumer dynamicFilterSourceConsumer = new LocalDynamicFilterConsumer(
                     dynamicFilterChannels,
                     dynamicFilterChannelTypes,
-                    ImmutableList.of(context.getCoordinatorDynamicFilterDomainsCollector(node.getDynamicFilters().keySet())),
+                    ImmutableList.of(taskContext::updateDomains),
```

Comment:
isn't this filter
```
.filter(entry -> coordinatorDynamicFilters.contains(entry.getKey()))
```
needed?

Reply:
In case of tardigrade, we can't have the optimisation of collecting from only one broadcast join task via `getCoordinatorDynamicFilters`. So all the DFs are collected and sent to coordinator, there is no need for filter, the `node.getDynamicFilters().keySet()` was there just to satisfy the method API.

Reply:
So this can cause regression in pipeline mode then if all tasks send their DFs.

Reply:
This code is in `visitDynamicFilterSource`, which is not used for pipeline mode

Reply:
That was the context I was missing. Could you add a comment about `DF collection from every task for FTE mode`  here?

Reply:
added comment